src/sinks/gcp/pubsub.rs

```diff
@@ -115,16 +115,18 @@ struct PubsubSink {
     encoding: EncodingConfigWithDefault<Encoding>,
 }
 
+const EMULATOR_HOST: &str = "http://localhost:8681";
+
 impl PubsubSink {
     async fn from_config(config: &PubsubConfig) -> crate::Result<Self> {
         // We only need to load the credentials if we are not targeting an emulator.
-        let creds = match config.emulator_host {
-            None => config.auth.make_credentials(Scope::PubSub).await?,
-            Some(_) => None,
+        let creds = match &config.endpoint {
+            Some(ref host) if host == EMULATOR_HOST => None,
```

Comment:
I'm not excited about hard-coding the emulator behavior this way. Yes, it works for the current specifics of our test environment, but will lead to unexpected changes in behavior if the details change (or if anybody else wants to target an emulator). Maybe a `skip_authentication: bool` in the config to control this behavior?

Reply:
yeah, this was existing behaviour.  it'd be good to keep the knowledge of specific hosts out of `from_config`
I've added `skip_authentication` as per your suggestion, thanks.  How to default this to `false` though?

Reply:
Instead of using `#[derive(default)]`, use this:
```rust
#[derive(Derivative)]
#[derivative(Default)]
struct PubsubConfig {
    …
    #[derivative(Default(value = "false"))]
    skip_authentication: bool,
}
```